CMakeLists.txt

```diff
@@ -3,9 +3,52 @@
 # See LICENSE.txt for the full license.
 
 cmake_minimum_required(VERSION 3.0)
+
+include("cmake/HunterGate.cmake")
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.16.15.tar.gz"
+    SHA1 "6974c2150fc0d3b09de3ad1efcbf15d360647ffa"
+)
+
 project(dlib VERSION 19.0)
 
-add_library(dlib INTERFACE)
+if(DLIB_HEADER_ONLY)
+  add_library(dlib INTERFACE)
+else()
+  
+  #duplicating dlib's default options config
+  if(DLIB_ISO_CPP_ONLY)
+    set(DLIB_JPEG_SUPPORT OFF)
+    set(DLIB_LINK_WITH_SQLITE3 OFF)
+    set(DLIB_USE_BLAS OFF)
+    set(DLIB_USE_LAPACK OFF)
+    set(DLIB_USE_CUDA OFF)
+    set(DLIB_PNG_SUPPORT OFF)
+    set(DLIB_GIF_SUPPORT OFF)
+  else()
+    set(DLIB_JPEG_SUPPORT ON)
+    set(DLIB_LINK_WITH_SQLITE3 ON)
+    set(DLIB_USE_BLAS ON)
+    set(DLIB_USE_LAPACK ON)
+    set(DLIB_USE_CUDA ON)
+    set(DLIB_PNG_SUPPORT ON)
+    set(DLIB_GIF_SUPPORT ON)
+  endif()
+
+  if(DLIB_JPEG_SUPPORT)
+    hunter_add_package(PNG CONFIG REQUIRED)
```

Comment:
Yes, so `hunter_add_package` should be called right before `find_package`. You now have:

```cmake
# top
if(DLIB_JPEG_SUPPORT)
  hunter_add_package(PNG)
endif()

# somewhere deeper in code
if(DLIB_JPEG_SUPPORT)
  find_package(PNG CONFIG REQUIRED)
endif()
```

This is a code duplication, why not simply:
```cmake
if(DLIB_JPEG_SUPPORT)
  hunter_add_package(PNG)
  find_package(PNG CONFIG REQUIRED)
endif()
```

?